ingest/ingest-publications.py

```diff
@@ -390,7 +391,39 @@ def generate(threads):
     g1 = g1 + describe(sparqlendpoint,get_pub_query)
     print("EMAIL: ", EMAIL)
 
+# section to create chunked files
+    chunk = 500
+    numchunks = 0
     records = generate(threads=int(args.threads))
-    print "generated records"
-    with open(args.out, "w") as bulk_file:
-      bulk_file.write('\n'.join(records))
+    rlen=len(records)
+    pubrecords = []
+    pubdoc={}
+    for i in range(rlen):
+        pubdata=json.loads(records[i])
+        pubmetadata=get_metadata(pubdata["pubId"])
+        #pubdoc.update({"pubmetadata:": pubmetadata})
+        #pubdoc.update({"pubdata:": pubdata})
+        pubrecords.append(json.dumps(pubmetadata))
+        pubrecords.append(json.dumps(pubdata))
+        numchunks += 1
+        dochunk = numchunks % chunk
+        if dochunk == 0:
```

Comment:
Any exception handling?

Reply:
I don't think I need error handling here, since this just appends a memory structure formatted in json to an array.
I could bake in some try/catch stuff to the file handlers that write, didn't yet because I would just get an error in the code and it would crash. That implies that the shell script should check for non-zero return codes and exit. You can see this at: https://github.com/cu-boulder/facetedsearch/pull/19/files#diff-61dadbe93985b15177b0d40977dc4a9bR10